cmd/crossplane/main.go

```diff
@@ -37,48 +36,57 @@ import (
 	"github.com/crossplane/crossplane/cmd/crossplane/rbac"
 )
 
+type debugFlag bool
+
+var cli struct {
+	Debug debugFlag `short:"d" help:"Print verbose logging statements."`
+
+	Core core.Command `cmd:"" help:"Start core Crossplane controllers." default:"1"`
+	Rbac rbac.Command `cmd:"" help:"Start Crossplane RBAC Manager controllers."`
+}
+
+func (d debugFlag) BeforeApply(ctx *kong.Context) error { // nolint:unparam
+	zl := zap.New(zap.UseDevMode(true))
+	ctx.BindTo(logging.NewLogrLogger(zl), (*logging.Logger)(nil))
+	ctrl.SetLogger(zl)
+	return nil
+}
+
 func main() {
-	var (
-		app   = kingpin.New(filepath.Base(os.Args[0]), "An open source multicloud control plane.").DefaultEnvars()
-		debug = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
-	)
-	c, ci := core.FromKingpin(app.Command("core", "Start core Crossplane controllers.").Default())
-	r, ri := rbac.FromKingpin(app.Command("rbac", "Start Crossplane RBAC Manager controllers."))
-	cmd := kingpin.MustParse(app.Parse(os.Args[1:]))
+	// // NOTE(negz): We must setup our logger after calling kingpin.MustParse in
+	// // order to ensure the debug flag has been parsed and set.
+	zl := zap.New(zap.UseDevMode(false))
+	ctrl.SetLogger(zl)
```

Comment:
This call was previously only made if the debug flag was true. Did you intend for it to be called now in all cases?

Reply:
No, I don't think so. I have removed this call from main, it should only be called in case of when the debug flag is passed, which is handled by the `BeforeApply` method. I removed your initial comments by mistake too, added them back with a new commit.